sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventProcessorClientWithStateManagement.java

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.messaging.eventhubs;
+
+import com.azure.messaging.eventhubs.models.CloseContext;
+import com.azure.messaging.eventhubs.models.CreateBatchOptions;
+import com.azure.messaging.eventhubs.models.ErrorContext;
+import com.azure.messaging.eventhubs.models.EventContext;
+import com.azure.messaging.eventhubs.models.InitializationContext;
+import com.azure.messaging.eventhubs.models.PartitionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.DirectProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+/**
+ * Demonstrates state management while processing events with {@link EventProcessorClient}. Shows usage of
+ * {@link EventProcessorClientBuilder#processPartitionInitialization(Consumer)},
+ * {@link EventProcessorClientBuilder#processError(Consumer)},
+ * {@link EventProcessorClientBuilder#processEvent(Consumer)}, and
+ * {@link EventProcessorClientBuilder#processPartitionClose(Consumer)}.
+ *
+ * An in-memory {@link CheckpointStore} is used to store checkpoint information for the sake of this demo. Production
+ * environments should leverage a durable store.
+ *
+ * A manufacturer has several machines on their assembly lines that emit temperature data. The manufacturer can use
+ * {@link EventProcessorClient} to aggregate the temperature data to look for anomalies. For example, the temperature
+ * data can say if a machine is over heating, or if no temperature data for a machine has been collected for a while, it
+ * may be offline.
+ *
+ * The partition key for each produced event is the name of the machine. This ensures that the temperature data for that
+ * machine always gets routed to the same partition. The contents of each event is the temperature of that machine in
+ * Celsius.
+ *
+ * Every 5 seconds, {@link MachineInformation} reports the current average temperature of the machine.
+ */
+public class EventProcessorClientWithStateManagement {
```

Comment:
While this sample is great and closer to real-world scenario, I am wondering if this is too big for users to grasp what is happening. In addition to this sample, having a simpler sample that shows how to maintain state would be a good first step before introducing to this sample.

Reply:
One of the things brought up during our stand up was to have a "real-world" use case of our EPH rather than the contrived printouts.

Reply:
I'll create one that just aggregates data from all partitions.